redwood-widget/src/iosMain/kotlin/app/cash/redwood/widget/RedwoodUIKitView.kt

```diff
@@ -53,6 +52,29 @@ public open class RedwoodUIKitView(
   }
 }
 
+public class RootUiView : UIView(cValue { CGRectZero }) {
```

Comment:
Working on migrating the Counter sample first sounds good.

I'm still trying to grok how `RedwoodView` would help us within a Redwood-only app, when `RedwoodComposition` is already (somewhat) solving the problem. Is the only benefit having `LocalUiConfiguration` set up? If so, I'm assuming the migration depends on finishing up the work to close https://github.com/cashapp/redwood/issues/1054?

If you have any psuedocode for what you think [what `onCreate` in `MainActivity`](https://github.com/cashapp/redwood/blob/trunk/samples/counter/android-views/src/main/kotlin/com/example/redwood/counter/android/views/MainActivity.kt) should look like, it'd be much appreciated!

Reply:
So there's a near-term and a long-term ideal.

For the near-term, I'd be happy with
```kotlin
class MainActivity : AppCompatActivity() {
  private val scope = CoroutineScope(AndroidUiDispatcher.Main)

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val redwoodView = RedwoodWidgetView(this)
    setContentView(redwoodView)

    val composition = RedwoodComposition(
      scope = scope,
      container = redwoodView.children,
      provider = SchemaWidgetFactories(
        Schema = AndroidWidgetFactory(this),
        RedwoodLayout = ViewRedwoodLayoutWidgetFactory(this),
      ),
    )
    composition.setContent {
      Counter()
    }
  }

  override fun onDestroy() {
    scope.cancel()
    super.onDestroy()
  }
}
```

Long term I think we want:
```kotlin
class MainActivity : AppCompatActivity() {
  private val scope = CoroutineScope(AndroidUiDispatcher.Main)
    super.onCreate(savedInstanceState)

    val redwoodView = RedwoodLayout(this)
    setContentView(redwoodView)

    redwoodView.setWidgetProvider(
      SchemaWidgetFactories(
        Schema = AndroidWidgetFactory(this),
        RedwoodLayout = ViewRedwoodLayoutWidgetFactory(this),
      )
    )
    redwoodView.setContent {
      Counter()
    }
  }
}
```

> Is the only benefit having `LocalUiConfiguration` set up? If so, I'm assuming the migration depends on finishing up the work to close https://github.com/cashapp/redwood/issues/1054?

This is the goal, yes. But the dependency is inverted. We need a standard mechanism of getting a `StateFlow<UiConfiguration>` before we can make it a required argument for `RedwoodComposition`.

Reply:
Sweet, thanks!

> We need a standard mechanism of getting a `StateFlow<UiConfiguration>`

Isn't the standard mechanism just `RedwoodView#uiConfiguration`?

Reply:
It will be, yes. But we need one of those for Android views, iOS UIKit, Compose UI, and the DOM and so far we only have about half sorta working. Once those implementations are done we can update `RedwoodComposition` to take a `RedwoodView` and automatically bind the composition local, thus completing #1054. And somewhere in there we can update the samples.